test/uk/gov/hmrc/residencenilratebandcalculator/models/CacheMapSpec.scala

```diff
@@ -0,0 +1,184 @@
+              )
+            )
+          )
+        )
+
+        val json = Json.toJson(complexData)
+
+        (json \ "id").as[String] mustBe "complex"
+        (json \ "data" \ "string").as[String] mustBe "value"
+        (json \ "data" \ "number").as[Int] mustBe 123
+        (json \ "data" \ "boolean").as[Boolean] mustBe true
+        (json \ "data" \ "array").as[Seq[String]] mustBe Seq("item1", "item2")
+        (json \ "data" \ "object" \ "nestedKey").as[String] mustBe "nestedValue"
+        (json \ "data" \ "object" \ "nestedArray").as[Seq[Int]] mustBe Seq(1, 2)
+      }
+
+      "successfully deserialize complex nested JSON" in {
+        val complexJson = Json.obj(
+          "id" -> "complex",
+          "data" -> Json.obj(
+            "string"  -> "value",
+            "number"  -> 123,
+            "boolean" -> true,
+            "array"   -> Json.arr("item1", "item2"),
+            "object" -> Json.obj(
+              "nestedKey"   -> "nestedValue",
+              "nestedArray" -> Json.arr(1, 2)
+            )
+          )
+        )
+
+        val result = complexJson.as[CacheMap]
+
+        result.id mustBe "complex"
+        result.data("string").as[String] mustBe "value"
+        result.data("number").as[Int] mustBe 123
+        result.data("boolean").as[Boolean] mustBe true
+        result.data("array").as[Seq[String]] mustBe Seq("item1", "item2")
+        result.data("object").as[JsObject].value("nestedKey").as[String] mustBe "nestedValue"
+        result.data("object").as[JsObject].value("nestedArray").as[Seq[Int]] mustBe Seq(1, 2)
+      }
+
+      "fail to deserialize when id is missing" in {
+        val invalidJson = Json.obj(
+          "data" -> Json.obj("key" -> "value")
+        )
+
+        intercept[JsResultException] {
+          invalidJson.as[CacheMap]
+        }
+      }
+
+      "fail to deserialize when data is missing" in {
+        val invalidJson = Json.obj(
+          "id" -> "missing-data"
+        )
+
+        intercept[JsResultException] {
+          invalidJson.as[CacheMap]
+        }
+      }
+
+      "fail to deserialize when data is not an object" in {
+        val invalidJson = Json.obj(
+          "id"   -> "invalid-data",
+          "data" -> "not an object"
+        )
+
+        intercept[JsResultException] {
+          invalidJson.as[CacheMap]
+        }
+      }
+
+      "handle round-trip conversion" in {
+        val original = CacheMap(
+          "round-trip",
+          Map("key1" -> JsString("value1"), "key2" -> JsNumber(42))
+        )
+
+        val json         = Json.toJson(original)
+        val deserialized = json.as[CacheMap]
+
+        deserialized mustBe original
+      }
+    }
+
+    "getEntry" should {
+      "return expected object" in {
+        model.getEntry[String]("key1") mustBe Some("value1")
+      }
+
+      "return None when missing key" in {
+        model.getEntry[String]("key_test") mustBe None
+      }
+      "throw an exception when object not matching" in {
```

Comment:
Please keep the spacing between tests.

The name of this test is clear about its intent. I suppose it should be rather something like:
`throw an exception when object **type** not matching`